junit-plugin/src/main/scala/org/scalajs/junit/plugin/ScalaJSJUnitPlugin.scala

```diff
@@ -0,0 +1,544 @@
+       *
+       * Example:
+       * {{{
+       * class Foo {
+       *   @Test def bar(): Unit
+       *   @Test def baz(): Unit
+       * }
+       * object Foo\$scalajs\$junit\$bootstrapper
+       *   // This is the method generated by mkInvokeJUnitMethodOnInstanceDef
+       *   def invoke(instance: AnyRef, methodName: String): Unit = {
+       *     if (methodName == "bar") instance.asInstanceOf[Foo].bar()
+       *     else if (methodName == "baz") instance.asInstanceOf[Foo].baz()
+       *     else throw new NoSuchMethodException(methodName + " not found")
+       *   }
+       * }
+       * }}}
+       */
+      def mkInvokeJUnitMethodOnInstanceDef(methods: List[MethodSymbol],
+          classSym: Symbol, refClassSym: Symbol): DefDef = {
+        val invokeJUnitMethodSym = classSym.newMethod(newTermName("invoke"))
+
+        val instanceParamSym =
+          invokeJUnitMethodSym.newValueParameter(newTermName("instance"))
+        instanceParamSym.setInfo(definitions.ObjectTpe)
+
+        val idParamSym =
+          invokeJUnitMethodSym.newValueParameter(newTermName("methodName"))
+        idParamSym.setInfo(definitions.StringTpe)
+
+        invokeJUnitMethodSym.setInfo(
+          MethodType(List(instanceParamSym, idParamSym), definitions.UnitTpe))
+
+        val methodNotFound = {
+          val msg = gen.mkMethodCall(idParamSym, nme.PLUS, Nil,
+            List(Literal(Constant(" not found"))))
+          val exception = mkNewInstance[NoSuchMethodException](List(msg))
+          Throw(exception)
+        }
+
+        val invokeJUnitMethodRhs0 = {
+          methods.foldRight[Tree](methodNotFound) {
+            case (methodSymbol, acc) =>
+              val headMethodId = Literal(Constant(methodSymbol.nameString))
+              val cond = gen.mkMethodCall(idParamSym, nme.equals_, Nil, List(headMethodId))
+              val instance = gen.mkAttributedIdent(instanceParamSym)
+              val castedInstance = gen.mkAttributedCast(instance, refClassSym.tpe)
+              val thenp = gen.mkMethodCall(castedInstance, methodSymbol, Nil, Nil)
+              If(cond, thenp, acc)
+          }
+        }
+
+        val invokeJUnitMethodRhs =
+          atOwner(invokeJUnitMethodSym)(typer.typed(invokeJUnitMethodRhs0))
+        val istanceParam = newValDef_compat(instanceParamSym, EmptyTree)()
+        val idParam = newValDef_compat(idParamSym, EmptyTree)()
+        typer.typedDefDef(newDefDef_compat(invokeJUnitMethodSym, invokeJUnitMethodRhs)(
+          vparamss = List(List(istanceParam, idParam))))
+      }
+
+      def mkGetJUnitMetadataDef(clSym: Symbol,
+          mlSymOption: Option[Symbol]): DefDef = {
+        val methods = jUnitAnnotatedMethods(clSym)
+        val mlMethods = mlSymOption.map(jUnitAnnotatedMethods)
+
+        def liftAnnotations(methodSymbol: Symbol): List[Tree] = {
+          val annotations = methodSymbol.annotations
+          def clone(t: Tree): Tree = {
```

Comment:
I found the source of the problem. Default parameters in the java annotations constructors are handled weirdly, guessing it is to make them follow that Java spces. By removing the default parameters in the `Test` I got rid of these cases, and the cloning :)